cmd/commands/cli/command.go

```diff
@@ -21,29 +21,61 @@ import (
 	"fmt"
 	"io"
 	"log"
+	"path/filepath"
 	"strconv"
 	"strings"
 
 	"github.com/chzyer/readline"
+	"github.com/mysteriumnetwork/node/cmd"
 	"github.com/mysteriumnetwork/node/metadata"
 	tequilapi_client "github.com/mysteriumnetwork/node/tequilapi/client"
 	"github.com/mysteriumnetwork/node/tequilapi/endpoints"
 	"github.com/mysteriumnetwork/node/utils"
+	"github.com/urfave/cli"
 )
 
-// NewCommand constructs CLI based with possibility to control quiting
-func NewCommand(
-	historyFile string,
-	tequilapi *tequilapi_client.Client,
-) *Command {
-	return &Command{
-		historyFile: historyFile,
-		tequilapi:   tequilapi,
+const cliCommandName = "cli"
+
+// NewCommand constructs CLI based Mysterium UI with possibility to control quiting
+func NewCommand() *cli.Command {
+	var di cmd.Dependencies
+
+	return &cli.Command{
+		Name:  cliCommandName,
+		Usage: "Starts a CLI client with a Tequilapi",
+		Action: func(ctx *cli.Context) error {
+			nodeOptions := cmd.ParseFlagsNode(ctx)
+			if err := di.Bootstrap(nodeOptions); err != nil {
+				return err
+			}
+			cmd.RegisterSignalCallback(utils.SoftKiller(di.Node.Kill))
+
+			errorChannel := make(chan error)
+			go func() {
+				if err := di.Node.Start(); err != nil {
+					errorChannel <- err
+					return
+				}
+				errorChannel <- di.Node.Wait()
+			}()
+
+			cmdCLI := &cliApp{
+				historyFile: filepath.Join(nodeOptions.Directories.Data, ".cli_history"),
+				tequilapi:   tequilapi_client.NewClient(nodeOptions.TequilapiAddress, nodeOptions.TequilapiPort),
+			}
+			cmd.RegisterSignalCallback(utils.HardKiller(cmdCLI.Kill))
```

Comment:
Node should be stopped also

Reply:
Added stop of the Node to the `After` function.